test/e2e/compatibility_test.go

```diff
@@ -105,3 +117,81 @@ query_tweaks:
+// NOTE: This requires dockerization of compliance framework: https://github.com/prometheus/compliance/pull/46
+func TestAlertCompliance_StatelessRuler(t *testing.T) {
+	t.Skip("This is an interactive test, using https://github.com/prometheus/compliance/tree/main/alert_generator testing is not optimized for CI runs")
+
+	e, err := e2e.NewDockerEnvironment("alert_compatibility")
+	testutil.Ok(t, err)
+	t.Cleanup(e.Close)
+
+	// Start receive + Querier.
+	receive, err := e2ethanos.NewIngestingReceiver(e, "receive")
+	testutil.Ok(t, err)
+	query := e2edb.NewThanosQuerier(e, "query_receive", []string{receive.InternalEndpoint("grpc")})
+
+	compliance := e.Runnable("alert_generator_compliance_tester").WithPorts(map[string]int{"http": 8080}).Init(e2e.StartOptions{
+		Image: "alert_generator_compliance_tester:latest",
+		// Batch job with HTTP port, we will start it with Exec, no need for readiness.
+		Readiness: alwaysReadyProbe{},
+		Command:   e2e.NewCommandWithoutEntrypoint("tail", "-f", "/dev/null"),
+	})
+	ruler := e2ethanos.NewStatelessRuler(e, "1", "rules", []alert.AlertmanagerConfig{
+		{
+			EndpointsConfig: httpconfig.EndpointsConfig{
+				StaticAddresses: []string{compliance.InternalEndpoint("http")},
+				Scheme:          "http",
+			},
+			Timeout:    amTimeout,
+			APIVersion: alert.APIv1,
+		},
+	}, []httpconfig.Config{
+		{
+			EndpointsConfig: httpconfig.EndpointsConfig{
+				StaticAddresses: []string{
+					query.InternalEndpoint("http"),
+				},
+				Scheme: "http",
+			},
+		},
+	}, []*config.RemoteWriteConfig{
+		{URL: &common_cfg.URL{URL: urlParse(t, e2ethanos.RemoteWriteEndpoint(receive.InternalEndpoint("remote-write")))}, Name: "thanos-receiver"},
+	})
+	testutil.Ok(t, e2e.StartAndWaitReady(receive, query, ruler, compliance))
+
+	// Pull rules.yaml:
+	{
+		var stdout bytes.Buffer
+		testutil.Ok(t, compliance.Exec(e2e.NewCommand("cat", "/rules.yaml"), e2e.WithExecOptionStdout(&stdout)))
+		testutil.Ok(t, os.MkdirAll(filepath.Join(ruler.Dir(), "rules"), os.ModePerm))
+		testutil.Ok(t, os.WriteFile(filepath.Join(ruler.Dir(), "rules", "rules.yaml"), stdout.Bytes(), os.ModePerm))
+	}
+
+	testutil.Ok(t, ioutil.WriteFile(filepath.Join(compliance.Dir(), "test-thanos.yaml"),
+		[]byte(alertCompatConfig(receive, query)), os.ModePerm))
+
+	fmt.Println(alertCompatConfig(receive, query))
+
+	testutil.Ok(t, compliance.Exec(e2e.NewCommand(
+		"/alert_generator_compliance_tester", "-config-file", filepath.Join(compliance.InternalDir(), "test-thanos.yaml")),
+	))
+}
+
+// nolint (it's still used in skipped test).
+func alertCompatConfig(receive e2e.Runnable, query e2e.Runnable) string {
+	return `settings:
+  remote_write_url: '` + e2ethanos.RemoteWriteEndpoint(receive.InternalEndpoint("http")) + `'
+  query_base_url: 'http://` + query.InternalEndpoint("http") + `'
+  rules_and_alerts_api_base_url: 'http://` + query.InternalEndpoint("http") + `'
```

Comment:
I don't see stateless ruler being set as one endpoint of the Thanos Querier so I guess we cannot read anything from the Querier? Maybe we can use ruler's endpoint directly?